Add unstructured grid for HR3/GFS

JessicaMeixner-NOAA opened the pull request:
<!-- PLEASE READ -->
<!-- Any PRs not following this template will be closed -->
<!--
    Please use a short (<60 char), descriptive title above. It should complete the sentence "If merged, this PR will _____". Capitalize the first word and do not end with a period.

    PRs should meet these guidelines:
    - Each PR should address ONE topic and have an associated issue.
    - No hard-coded paths or personal directories.
    - No temporary or backup files should be committed (including logs).
    - Any code that you disabled by being commented out should be removed or reenabled.

    Please delete all these comments before submitting the PR.
-->
# Description
<!-- This description will become the commit message for the PR-->
<!-- Please use this format for your description:

  Describe your changes. Focus on the *what* and *why*. The *how* will be evident from the changes. In particular, be sure to note any interface changes, such as command line syntax, that will need to be communicated to users.

  At the end of your description, please be sure to add the issue this PR solves using the word "Resolves". If there are any issues that are related but not yet resolved (including in other repos), you may use "Refs".

  Resolves #1234
  Refs #4321
  Refs NOAA-EMC/repo#5678
-->

This adds the capability to use unstructured grids in the global workflow, which will be used in HR3.  There are new fix files for a low-resolution 100km grid and a grid closer to our targeted GFSv17 grid which has the resolution combined from the older multi_1 and GFSv16 grids.    The fix file update is here: https://github.com/NOAA-EMC/global-workflow/issues/2229 

Note: This now means that GFS tests need a new build option: 

./build_all.sh -w 
So that PDLIB=ON is turned on for compiling relevant UFS and WW3 codes. 


This Closes https://github.com/NOAA-EMC/global-workflow/issues/1547 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES/NO - We'll need to make sure all necessary tests are updated or added to.  
- Does this change require a documentation update? YES - added build instructions 

# How has this been tested?

I've tested this on orion, hera and wcoss2 during various stages of development.  

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes -- New tests are needed, I just need to make sure I coordinate appropriately with code-managers.  A low resolution grid is provided, just need ICs for appropriate days // to build in flexiblity for unstructured vs structured as desired by CMs.
- [x] I have made corresponding changes to the documentation if necessary

JessicaMeixner-NOAA committed:
update code to a version that has unstructured grid features

JessicaMeixner-NOAA committed:
commit local changes from a few weeks ago

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
updates to get wave prep to run

JessicaMeixner-NOAA committed:
change name of unstructured grid to be short enough for ww3

JessicaMeixner-NOAA committed:
update to s2s default

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
update to latest ufs weather model

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

 Conflicts:
	sorc/build_all.sh
	sorc/build_ufs.sh
	sorc/build_ww3prepost.sh

JessicaMeixner-NOAA committed:
update version of ufs to have 7.0.4 of scotch on RDHPCS

JessicaMeixner-NOAA committed:
update commit hash of ufs-weather-model

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

Conflicts:
	Externals.cfg
	sorc/checkout.sh

JessicaMeixner-NOAA committed:
updates for build

JessicaMeixner-NOAA committed:
temp file

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
updates for new unstructured grids

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
use UFS that has scotch 7.0.4 on WCOSS2

JessicaMeixner-NOAA committed:
temporary files added

JessicaMeixner-NOAA committed:
updates to run with C768 on wcoss2

JessicaMeixner-NOAA committed:
updates for wave configuration

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

 Conflicts:
	Externals.cfg
	sorc/build_all.sh
	sorc/checkout.sh

JessicaMeixner-NOAA committed:
update ufs model to code that was used in previous commit

JessicaMeixner-NOAA committed:
orion temp update

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
updates

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
revert bypass of wave IC

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwave

JessicaMeixner-NOAA committed:
clean up for PR

JessicaMeixner-NOAA committed:
attempt to fix shellcheck

WalterKolczynski-NOAA commented:
Is there a reason to not always build with PDLIB on?

JessicaMeixner-NOAA committed:
shell check updates round 2

JessicaMeixner-NOAA commented:
If you would like to use structured grids, like GEFS, you have to build with this off.

JessicaMeixner-NOAA committed:
shell check round 3

WalterKolczynski-NOAA commented:
> If you would like to use structured grids, like GEFS, you have to build with this off.

Okay, this is going to be a bit bigger change then. Current CI builds once and uses it for all tests, but that won't be possible if we need a different executable for different tests. The CI tests aren't being updated here, so not an issue yet, but will be one if an unstructured grid is later added as a CI test (which it should, if the HR3 configuration sticks for GFS v17).

@TerrenceMcGuinness-NOAA @aerorahul

JessicaMeixner-NOAA commented:
I have promised @aerorahul corresponding tests and I have updated things in the config.base from their previous grids so that will also have impacts on the existing tests.  The details of how to get the tests in are not somehting I felt comfortable trying to tackle as there's not an option for how to specify a build option --- yet!   But I want to make sure I provide anything and everythign to make that happen.

JessicaMeixner-NOAA committed:
update for shellcheck

JessicaMeixner-NOAA committed:
more shell check update attempts

aerorahul commented:
Thanks @JessicaMeixner-NOAA 
Yes. Associated tests with the unstructured grid will be appreciated.

However, there is another issue here besides the new tests that is coming to light.
Presently, we only build one version of the model, WW3 utilities and they are shared for all tests; GFS and GEFS.
If I understand correctly, since WW3 is not runtime configurable, this will need compiling the model and WW3 utilities differently and the right one will have to be made available depending on the configuration and the system.
To do this, multiple builds will need to be introduced, and the testing system will need to be updated.

JessicaMeixner-NOAA commented:
> Thanks @JessicaMeixner-NOAA Yes. Associated tests with the unstructured grid will be appreciated.
> 
> However, there is another issue here besides the new tests that is coming to light. Presently, we only build one version of the model, WW3 utilities and they are shared for all tests; GFS and GEFS. If I understand correctly, since WW3 is not runtime configurable, this will need compiling the model and WW3 utilities differently and the right one will have to be made available depending on the configuration and the system. To do this, multiple builds will need to be introduced, and the testing system will need to be updated.

Yes, this is correct @aerorahul

JessicaMeixner-NOAA commented:
If there's a desire to do things in stages because of the complexity of the build system additional requirements, I can un-do the updates for the lower resolution wave grids to default to structured grids.  Then once its possible to have both, we can update to have an unstructured test.  Just let me know what is the preference and I can provide anything needed for a low resolution unstructured grid test for the workflow.

aerorahul commented:
> If there's a desire to do things in stages because of the complexity of the build system additional requirements, I can un-do the updates for the lower resolution wave grids to default to structured grids. Then once its possible to have both, we can update to have an unstructured test. Just let me know what is the preference and I can provide anything needed for a low resolution unstructured grid test for the workflow.

Doing things in stages makes sense to ensure that the new features added as part of this PR are tested and previous required capabilities are retained. Now that we are aware of these requirements, we can begin to scope out the work and the time needed to achieve this as quickly as possible.
We can further discuss this offline.

JessicaMeixner-NOAA committed:
revert two files so that existing tests will work

JessicaMeixner-NOAA commented:
Updates have been made so that existing tests will work.  Should I make any follow-up issues to track the need for new tests for the unstructured grid?

JessicaMeixner-NOAA committed:
since tests only go up to C384, setting C768/C1152 to desired unstructred grid as default

WalterKolczynski-NOAA commented on sorc/build_ufs.sh:
```diff
@@ -7,12 +7,13 @@ cwd=$(pwd)
     a) APP="${OPTARG}";;
     j) BUILD_JOBS="${OPTARG}";;
     v) export BUILD_VERBOSE="YES";;
+    w) export PDLIB="ON";; 
```
I don't think this needs to be exported, since it is just turned into a cmake argument in this script
```suggestion
    w) PDLIB="ON";; 
```

JessicaMeixner-NOAA replied:
I will test this change before committing it to the branch.

WalterKolczynski-NOAA commented on scripts/exgfs_wave_init.sh:
```diff
@@ -156,17 +163,17 @@ source "${HOMEgfs}/ush/preamble.sh"
     then
       if [ ${CFP_MP:-"NO"} = "YES" ]; then
-        ${wavempexec} -n ${wavenproc} ${wave_mpmd} cmdfile
+        "${wavempexec} -n ${wavenproc} ${wave_mpmd} cmdfile"
```
```suggestion
        ${wavempexec} -n "${wavenproc}" "${wave_mpmd}" cmdfile
```

WalterKolczynski-NOAA commented on scripts/exgfs_wave_init.sh:
```diff
@@ -156,17 +163,17 @@ source "${HOMEgfs}/ush/preamble.sh"
+        "${wavempexec} -n ${wavenproc} ${wave_mpmd} cmdfile"
       else
-        ${wavempexec} ${wavenproc} ${wave_mpmd} cmdfile
+        "${wavempexec} ${wavenproc} ${wave_mpmd} cmdfile"
```
```suggestion
        ${wavempexec} "${wavenproc}" "${wave_mpmd}" cmdfile
```

WalterKolczynski-NOAA commented on scripts/exgfs_wave_init.sh:
```diff
@@ -118,6 +118,13 @@ source "${HOMEgfs}/ush/preamble.sh"
+
+      if [ -f ${FIXwave}/${grdID}.msh ]
+      then
+        cp $FIXwave/${grdID}.msh ${grdID}.msh
```
```suggestion
        cp "${FIXwave}/${grdID}.msh" "${grdID}.msh"
```

WalterKolczynski-NOAA commented on scripts/exgfs_wave_init.sh:
```diff
@@ -118,6 +118,13 @@ source "${HOMEgfs}/ush/preamble.sh"
+      then
+        cp $FIXwave/${grdID}.msh ${grdID}.msh
+      fi
+      #TO DO: how do we say "it's unstructured, and therefore need to have error check here" 
```
What kind of error check are you thinking of?

JessicaMeixner-NOAA replied:
We currently don't have a variable that says unstructured or not.  If it is unstructured and you don't have this file, you'll fail in the execution of ww3_grid a few lines below, but perhaps this should be caught here instead?

JessicaMeixner-NOAA replied:
@WalterKolczynski-NOAA any requests for changes here?  Otherwise, I think I have addressed all of your other comments.

WalterKolczynski-NOAA replied:
I think this is good enough for now.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
Update scripts/exgfs_wave_init.sh

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

JessicaMeixner-NOAA committed:
Update scripts/exgfs_wave_init.sh

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

JessicaMeixner-NOAA committed:
Update scripts/exgfs_wave_init.sh

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

JessicaMeixner-NOAA committed:
Update sorc/build_ufs.sh

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

WalterKolczynski-NOAA reviewed (approved):
Conditionally approved pending successful completion of CI tests.

JessicaMeixner-NOAA commented:
I'll look at the failed logs today.  I'm wondering if they are related to some of the changes made for the shell-check.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA commented:
I have confirmed that the failures were caused by the changes required to pass the shell-check.

JessicaMeixner-NOAA committed:
fix reason of ci test failures

JessicaMeixner-NOAA commented:
The code now fails shellnorms, however I'm unaware of a way to pass shell norms and have the code succeed.  Moreover, it's on a line of code that was not originally changed. Please let me know if changing this unrelated code is going to be required of this PR.    I'm continuing to run the rest of the test, but we should at least be past the first point of failure of the CI tests from last night.

WalterKolczynski-NOAA commented:
@JessicaMeixner-NOAA Just ignore it if you didn't change it. We'll get it later.

JessicaMeixner-NOAA commented:
Thanks @WalterKolczynski-NOAA.  My test is on the last bit of post jobs, so it's likely okay to re-start the tests for ensuring structured grid was not unintentionally broken with adding in unstructured grid.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA commented:
* The re-test of the structured grid case succeeded: /work2/noaa/marine/jmeixner/HR3/unstrucPRstructest/test01 
* The re-test of the unstructured low-res case (non-default options), also succeeded: /work2/noaa/marine/jmeixner/HR3/unstrucPR/utest02
* I also re-confirmed that the model does not run a structured grid when compiled with PDLIB=ON.

JessicaMeixner-NOAA commented:
@aerorahul @WalterKolczynski-NOAA   I've been thinking more about some ways we could add in the extra build and wanted to check to see if you already had ideas etc.  While it's not garunteed, for now the ww3 executables should be the same, although we'd have to confirm, but then we could just have a second forecast executable with PDLIB=ON, and then add a variable for "yes unstructured" which will then say which forecast exe to use.   Or there might be other solutions as well.  Please let me know how I can help!

JessicaMeixner-NOAA commented:
Actually thinking more, we could do the same for the WW3 executables as well and just future-proof it.

WalterKolczynski-NOAA commented:
@JessicaMeixner-NOAA We've already decided on a design and Terry is working to implement it this week. I believe this PR is being held for that.

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA Thanks for communicating the status.  Is there a desire to ensure the structured grid tests work now or not?    Do I understand correctly that the timeline for the update to the tests in this week?

WalterKolczynski-NOAA commented:
Yes on tests, and timeline is we're doing our best to get the framework in place this week.

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA  Thanks again for the update and to you @aerorahul and @TerryMcGuinness-NOAA  for helping get the testing in place.  Please let me know if there's anything I can do to help.

WalterKolczynski-NOAA commented:
@JessicaMeixner-NOAA Terry has finished the work getting the dual build system in and we've merged it (#2246), so you can start adding the modifications you need (we may need a bit of time to do a shakedown of getting the system in place for the main repo). You will need to update `ci/cases/yamls/build.yaml` with the new build signature for gfs, then make sure all the gfs resolutions use an unstructured grid.

JessicaMeixner-NOAA commented:
Thanks for the update @WalterKolczynski-NOAA !  I'll get this branch sync'd up with develop and start working on the test ASAP!

JessicaMeixner-NOAA committed:
lower task count so it runs out of box on hera

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
update build for unstructured waves

JessicaMeixner-NOAA commented:
Quick question on the CI tests - Right now included with the workflow fix files I have a low resolution grid for testing which is ~100km and is the smallest grid I have, and then I have the currently targeted grid which is much higher resolution.   For The C384 weekly CI test should I point: 
-- The low resolution 100km unstructured grid 
-- The high resolution grid used at C768 and in HRs 
-- Add additional fix files for ~20km unstructured grid. 

@aerorahul @WalterKolczynski-NOAA   - any preferences?  I'm leaning towards the 100km since I still think you want this to be a fairly lightweight test.

JessicaMeixner-NOAA committed:
all gfs wave grids need to be unstructured since we build with -w

JessicaMeixner-NOAA commented:
My test is still going but it's far enough along that I think we can start to move forward.  

The following two ICs need to be copied to the main IC areas: 
/scratch1/NCEPDEV/climate/Jessica.Meixner/HR3/updatetest/IC/workflow_C384_refactored/2016070100/wave/20160701.000000.restart.uglo_100km 
to $ICDIR/workflow_C384_refactored/2016070100/wave/20160701.000000.restart.uglo_100km 

and 
/scratch1/NCEPDEV/climate/Jessica.Meixner/HR3/updatetest/IC/workflow_C48_refactored/2021032312/wave/20210323.120000.restart.uglo_100km 
to 
$ICDIR/workflow_C48_refactored/2021032312/wave/20210323.120000.restart.uglo_100km 

These are not perfect ICs, but should be sufficient for testing. After these ICs are copied into the various locations I believe this is ready for CI testing again.  Please let me know if there are any other updates I should make or tests you'd like to see before running CI tests.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

WalterKolczynski-NOAA commented:
```
24:        Type 8 : Restart files second request  
24:       -----------------------------------------
24:             From     : 2021/03/23 12:00:00 UTC
24:             To       : 2021/03/28 12:00:00 UTC
24:             Interval :            12:00:00
24: 
24:        Wave model ...
24:  EXPTOTAL is selected
24:  But PDLIB is not
55:  EXPTOTAL is selected
55:  But PDLIB is not
40:  EXPTOTAL is selected
40:  But PDLIB is not
41:  EXPTOTAL is selected
41:  But PDLIB is not
42:  EXPTOTAL is selected
42:  But PDLIB is not
43:  EXPTOTAL is selected
43:  But PDLIB is not
45:  EXPTOTAL is selected
45:  But PDLIB is not
46:  EXPTOTAL is selected
46:  But PDLIB is not
47:  EXPTOTAL is selected
47:  But PDLIB is not
48:  EXPTOTAL is selected
48:  But PDLIB is not
49:  EXPTOTAL is selected
49:  But PDLIB is not
50:  EXPTOTAL is selected
50:  But PDLIB is not
51:  EXPTOTAL is selected
51:  But PDLIB is not
52:  EXPTOTAL is selected
52:  But PDLIB is not
53:  EXPTOTAL is selected
53:  But PDLIB is not
54:  EXPTOTAL is selected
54:  But PDLIB is not
56:  EXPTOTAL is selected
56:  But PDLIB is not
57:  EXPTOTAL is selected
57:  But PDLIB is not
58:  EXPTOTAL is selected
58:  But PDLIB is not
59:  EXPTOTAL is selected
59:  But PDLIB is not
60:  EXPTOTAL is selected
60:  But PDLIB is not
61:  EXPTOTAL is selected
61:  But PDLIB is not
62:  EXPTOTAL is selected
62:  But PDLIB is not
63:  odat(j=4):     20210324           0       43200    20210328      120000
63:  EXPTOTAL is selected
63:  But PDLIB is not
 0: slurmstepd: error: *** STEP 16565725.0 ON Orion-01-72 CANCELLED AT 2024-02-13T02:00:22 DUE TO TIME LIMIT ***
slurmstepd: error: *** JOB 16565725 ON Orion-01-72 CANCELLED AT 2024-02-13T02:00:22 DUE TO TIME LIMIT ***
```

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA the error above to me looks like the model was not built with the -w option.

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA I have confirmed that the build was not with the PDLIB option in the log here: /work2/noaa/stmp/GFS_CI_ROOT/ORION/PR/2230/global-workflow/sorc/logs/build_ufs.log   

Is there a log of what was called to do this build?

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

WalterKolczynski-NOAA commented:
Oops, this was a brain fart on my part. We haven't switched over to the CI capable of multi-build on Orion yet.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
increased memory was required on wcoss2 for scout runs

JessicaMeixner-NOAA committed:
update to HR3 ICs

WalterKolczynski-NOAA commented:
We're still having issues with Jenkins, so we're going to plan B: disable the GEFS test for now and stay on the shell CI. I will make a be opening a PR into this PR shortly.

WalterKolczynski-NOAA committed:
Update shell-based CI scripts

Updates the shell-based CI scripts to build with the new `-w` option.

The GEFS test is temporarily disabled pending switch to the Jenkins
system and multi-build capabililty in the CI system.

WalterKolczynski-NOAA commented:
PR opened into this PR: https://github.com/JessicaMeixner-NOAA/global-workflow/pull/9

@JessicaMeixner-NOAA if you can merge that in, we'll get the testing done so we can finally get this in.

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA thanks for the information.  I have merged your PR to my branch.  Since  I knew from @aerorahul yesterday that this was taking a bit longer than expected, I was starting to collect the other HR3 updates here so we'd just have one PR and be ready to go (hopefully).  I updated the pointers to the ICs, but I'm not sure if the files have been linked yet: https://github.com/NOAA-EMC/global-workflow/issues/2314 Also we will need https://github.com/NOAA-EMC/global-workflow/issues/2316 for HR3 and I can point to this new fix here or we can wait and do that as a follow up PR. 

Please let me know what I need to do from my end and thanks for the information and all of the efforts towards getting the multi-build up and running.  I had thought about another plan which would be to build the UFS and WW3 twice and put "pdlib" in the exe names and update that, which could be an option to follow through on if for whatever reason the Jenkins takes longer than expected or there are unforeseen complications.  This could be a stop-gap or potential longer term solution.  Happy to discuss further if there's interest.

WalterKolczynski-NOAA commented:
@JessicaMeixner-NOAA ICs have been linked on all three machines. @KateFriedman-NOAA has fix files in place on Hera and is just waiting on confirmation from Mike before synchronizing.

JessicaMeixner-NOAA committed:
updating ugwd fix file

JessicaMeixner-NOAA committed:
Merge branch 'feature/unstructuredwavePR' of github.com:JessicaMeixner-NOAA/global-workflow into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/unstructuredwavePR

WalterKolczynski-NOAA commented:
@JessicaMeixner-NOAA We need to make one quick change:

Change `hera.intel` to `hera` at https://github.com/NOAA-EMC/global-workflow/blob/develop/sorc/build_all.sh#L132

JessicaMeixner-NOAA commented:
Thanks @WalterKolczynski-NOAA I was looking inot why this was failing.. I just sync'd up my branch to see if I was missing something.  I'll add this now.  I'm assuming this will fix the reason that gdas app will not build?

WalterKolczynski-NOAA commented:
> Thanks @WalterKolczynski-NOAA I was looking inot why this was failing.. I just sync'd up my branch to see if I was missing something. I'll add this now. I'm assuming this will fix the reason that gdas app will not build?

Yes.

JessicaMeixner-NOAA committed:
hera.intl to hera in gdas build all

JessicaMeixner-NOAA commented:
@WalterKolczynski-NOAA hera.intel to hera update is done!

JessicaMeixner-NOAA commented:
I've looked into: /work2/noaa/stmp/GFS_CI_ROOT/ORION/PR/2230/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_799d942f/logs/2021032418/gdasocnanalprep.log and it's because we do not have obs on orion and are pointing to a directory in hera.

WalterKolczynski-NOAA commented:
That test shouldn't be running in the first place. There's a typo in the case file. I sent the correction to you in slack.

JessicaMeixner-NOAA committed:
fix typo, to turn off C48mx500_3DVarAOWCDA tests
